R/dataframe-group_by-general.R

```diff
@@ -35,11 +36,18 @@ wrap_to_group_by <- function(x, by, maintain_order) {
 #'   c_mean_squared = (pl$col("c") ** 2)$mean()
 #' )
 groupby__agg <- function(...) {
-  self$df$lazy()$group_by(
-    !!!self$by,
-    .maintain_order = self$maintain_order
-  )$agg(...)$collect(optimizations = QueryOptFlags()$no_optimizations()) |>
-    wrap()
+  wrap({
+    out <- self$df$lazy()$group_by(
+      !!!self$by,
+      .maintain_order = self$maintain_order
+    )
+
+    if ("predicates" %in% names(self)) {
```

Comment:
Maybe `if (length(self$predicates) > 0L)`?

Reply:
Can't check with the code right now but the issue was that `$.polars_object` would error. But maybe that changed with your latest commit, I'll double check later and add the rolling/dynamic groupby.